### PR TITLE
Adds initial memory reads and some performance fixes

### PR DESCRIPTION
**Pull request by Eein**
## Memory Core

This was built with the notion that the game may not be running, or the game my crash, or we may need to close the game to reset some type of RNG

Theres two ways to use it, either by using `get_pointer` or `follow_pointer` then using `read_XXX` of whatever type from `pymem`

- `get_pointer` gets the pointer how people are used to, however when we want to store a base addr for performance (see `PlayerPartyManager`, where we dont want to read 30 memory addresses over and over) this differs: 
- `follow_pointer` is used to start from where you left off and only make the necessary calls.
- to point towards a bit of memory and just offset, you can end follow with 0x0 and read directly with `ptr + 0xXX` (see `position`)

# Changelog
- adds a memory core
  - this core should not crash the app when the game crashes, nor should it crash the app if the game is not open. Values returned should be None/Default (this can be mulled over later)
- adds memory/PlayerPartyManager - this handles coordinates for now
- adds a `Vec` class I found somewhere on github
- adds a `time.sleep` to the update loop so my graphics card doesnt have to render 35 million frames a second and turn my room into a hellscape (if anyone knows how to do this in imgui, please do)
- adds numpy, vgamepad, pymem to requirements.txt
- adds coordinates/position to imgui GUI
- I dont know what `__init__.py` does - but i added one anyways

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -1,2 +1,5 @@
 imgui[glfw]==2.0.0
 pyyaml==6.0
+numpy
+pymem
+vgamepad
```
Comment (orkaboy) [inline requirements.txt:5]: Woops, forgot to add this one...

File: helpers/vec.py [hunk 2]
```diff
@@ -0,0 +1,54 @@
+import math
+
+# https://gist.github.com/tatsy/e14dd18079bca60ac8f78217b77332c1
+
+class Vec(object):
```
Comment (orkaboy) [inline helpers/vec.py:5]: I already added something similar to this (a Vec2) in engine/mathlib (though currently unused). They should probably be in the same place at least.

File: engine/seq/sequencer.py [hunk 3]
```diff
@@ -59,10 +60,15 @@ def _get_deltatime(self) -> float:
         return delta
 
     def _update(self) -> None:
+        time.sleep(0.008333333)
```
Comment (orkaboy) [inline engine/seq/sequencer.py:63]: Is this one needed? There will be some sleep due to the gui polling/event handling. Either way, should probably not be a magic number.

File: memory/player_party_manager.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import memory.core as Memory
```
Comment (orkaboy) [inline memory/player_party_manager.py:1]: Prefer `from memory.core import mem_handle`

File: memory/player_party_manager.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+    def update(self):
+        self.memory.update()
+
+        if self.memory is not None:
+            offset = [
+                0xB8, 0x10, 0x40, 0x30, 0x10, 0x38, 0xA8, 0x10, 
+                0xC8, 0x38, 0xE8, 0x20, 0x40, 0xD0, 0x200
+            ] 
+            ptr = self.memory.get_pointer(0x2EAA3D0, offset)
```
Comment (orkaboy) [inline memory/player_party_manager.py:17]: Prefer putting this as a named constant (the address).

File: engine/seq/sequencer.py (hunk 3 above)
Comment (Eein) [reply engine/seq/sequencer.py:63]: this is "basically" 120fps, without it, python burns a core to the ground, and my gpu is rendering half a trillion frames 
glad to change this.

File: memory/player_party_manager.py (hunk 5 above)
Comment (Eein) [reply memory/player_party_manager.py]: just noting here - but i had an issue where moving it as a constant may see it show up initialized as an empty list at times, causing issues.

**Pull request merged**